tests/vm_console_proxy_test.go

```diff
@@ -0,0 +1,188 @@
+			Namespace:      strategy.GetVmConsoleProxyNamespace(),
+			Resource:       &apps.Deployment{},
+			ExpectedLabels: expectedLabels,
+			UpdateFunc: func(deployment *apps.Deployment) {
+				deployment.Spec.Replicas = pointer.Int32(0)
+			},
+			EqualsFunc: func(old *apps.Deployment, new *apps.Deployment) bool {
+				return reflect.DeepEqual(old.Spec, new.Spec)
+			},
+		}
+		configMapResource = testResource{
+			Name:           "vm-console-proxy",
+			Namespace:      strategy.GetVmConsoleProxyNamespace(),
+			Resource:       &core.ConfigMap{},
+			ExpectedLabels: expectedLabels,
+			UpdateFunc: func(configMap *core.ConfigMap) {
+				configMap.Data = map[string]string{"tls-profile-v1alpha1.json": "{}"}
+			},
+			EqualsFunc: func(old *core.ConfigMap, new *core.ConfigMap) bool {
+				return reflect.DeepEqual(old.Data, new.Data)
+			},
+		}
+
+		waitUntilDeployed()
+	})
+
+	AfterEach(func() {
+		strategy.RevertToOriginalSspCr()
+	})
+
+	Context("Resource creation", func() {
+		DescribeTable("created cluster resource", func(res *testResource) {
+			resource := res.NewResource()
+			err := apiClient.Get(ctx, res.GetKey(), resource)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(hasOwnerAnnotations(resource.GetAnnotations())).To(BeTrue(), "owner annotations are missing")
+		},
+			Entry("[test_id:TODO] cluster role", &clusterRoleResource),
+			Entry("[test_id:TODO] cluster role binding", &clusterRoleBindingResource),
+			Entry("[test_id:TODO] service account", &serviceAccountResource),
+			Entry("[test_id:TODO] service", &serviceResource),
+			Entry("[test_id:TODO] deployment", &deploymentResource),
+			Entry("[test_id:TODO] config map", &configMapResource),
+		)
+
+		DescribeTable("should set app labels", expectAppLabels,
+			Entry("[test_id:TODO] cluster role", &clusterRoleResource),
+			Entry("[test_id:TODO] cluster role binding", &clusterRoleBindingResource),
+			Entry("[test_id:TODO] service account", &serviceAccountResource),
+			Entry("[test_id:TODO] service", &serviceResource),
+			Entry("[test_id:TODO] deployment", &deploymentResource),
+			Entry("[test_id:TODO] config map", &configMapResource),
+		)
+	})
+
+	Context("Resource deletion", func() {
+		DescribeTable("recreate after delete", expectRecreateAfterDelete,
+			Entry("[test_id:TODO] cluster role", &clusterRoleResource),
+			Entry("[test_id:TODO] cluster role binding", &clusterRoleBindingResource),
+			Entry("[test_id:TODO] service account", &serviceAccountResource),
+			Entry("[test_id:TODO] service", &serviceResource),
+			Entry("[test_id:TODO] deployment", &deploymentResource),
+			Entry("[test_id:TODO] config map", &configMapResource),
+		)
+	})
+
+	Context("Resource change", func() {
+		DescribeTable("should restore modified resource", expectRestoreAfterUpdate,
+			Entry("[test_id:TODO] cluster role", &clusterRoleResource),
+			Entry("[test_id:TODO] cluster role binding", &clusterRoleBindingResource),
+			Entry("[test_id:TODO] service account", &serviceAccountResource),
```

Comment:
The update test for service account currently does not do any update, so it can be removed.

Reply:
Also in `should restore modified resource with pause`, right?

Reply:
yes